Rewrite to use current Box API

robotski opened the pull request.

robotski committed:
Rewrite to use current API version

(yeah I know big blob commits like this are bad.)

robotski committed:
Set version to 2.0.0

robotski commented on ua_box_api/ua_box_api.py:
```diff
@@ -9,69 +8,84 @@
+        [result] = [item for item in search.entries if item.name == name]
+        return result
+
+        # This block would be reasonable for the first of one or more matches.
+        # Could also do all results? Then we could use get_duplicate_file_name.
+        # But now we'd be returning a list instead of a
+        # FileFull | FolderFull | WebLink | SearchResultWithSharedLink
+
+        # result = next((item for item in search.entries if item.name == name), None)
+        # if result:
+        #     return result
+        # else:
+        #     raise FileNotFoundError("Failed to find requested item.")
```
Not sure if any of these would be better than what I went with.

sterns1 commented on ua_box_api/ua_box_api.py:
```diff
@@ -9,69 +8,84 @@
+        self.client = BoxClient(auth=auth)
 
-    def get_all_items(self, item_id):
+    def get_all_items(self, item_id: int | str):
```
I know it's only because that's how it was written, but I can't stand OR or Optional's in type hints. We are str'ing it below, so we can just say str here. And can you add a return type hint too?

sterns1 replied:
Or to be more pythonic, I just wouldn't str it at all. Say str in the type hint, and it's on the caller if they give something weird.

sterns1 commented on ua_box_api/ua_box_api.py:
```diff
@@ -9,69 +8,84 @@
+            return items
 
-    def find_child_by_name(self, name, item_id):
+    def find_child_by_name(self, name: str, item_id: int | str):
```
Same peice about the item_id hints and the return type hint

sterns1 commented on ua_box_api/ua_box_api.py:
```diff
@@ -9,69 +8,84 @@
-    def get_duplicate_file_name(self, folder_id, current_name, zip_file=False):
+        # This construct throws a ValueError if there isn't exactly 1 match.
+        # https://stackoverflow.com/a/7008062
+        # https://dbader.org/blog/python-nested-unpacking
```
This kind of documentation is distracting. Should we be returning a collection of matches? If you need to add stackoverflow links, I'd suggest using ADO for that.

robotski committed:
Update type hints

Remove optional types in parameters, ids are only str now.

Remove str conversion for item_id in get_all_items, no longer needed.

Add return type hints and necessary imports.

robotski committed:
Update comments

Remove commented out code.

Replace the StackOverflow links in find_child_by_name with an actual
description of how unpacking a list comprehension into result works.

Fix outdated and wonky comments.

robotski committed:
Enforce desired search_for_content behavior

The include_recent_shared_links parameter of the Box API's
search_for_content is False by default. Explicitly setting it prevents
potential undesired future behavior.
